extensions/blocks/tiled-gallery/tiled-gallery.php

```diff
@@ -7,28 +7,121 @@
  * @package Jetpack
  */
 
-if (
-	( defined( 'IS_WPCOM' ) && IS_WPCOM ) ||
-	class_exists( 'Jetpack_Photon' ) && Jetpack::is_module_active( 'photon' )
-) {
-	jetpack_register_block(
-		'jetpack/tiled-gallery',
-		array(
-			'render_callback' => 'jetpack_tiled_gallery_load_block_assets',
-		)
-	);
+/**
+ * Jetpack Tiled Gallery Block class
+ *
+ * @since 7.4
+ */
+class Jetpack_Tiled_Gallery_Block {
+	/* Values for building srcsets */
+	const IMG_SRCSET_WIDTH_MAX  = 2000;
```

Comment:
We might need to elaborate about Photon's 2000px limit here to explain reasoning for this value.

Reply:
Maybe even mention it should be kept in sync with JS side and vice-versa?

https://github.com/Automattic/jetpack/blob/0721157d9dc376697de31c1cd5a845622efaba7a/extensions/blocks/tiled-gallery/constants.js#L5

Reply:
Photon doesn't have a pixel limit, it has a file size limit. `2000px` was a size I observed to work well with photon, but we don't really have a way of knowing in JavaScript what to expect.

These sizes don't really need to stay in sync, this is just the maximum size we'll use in the srcset 🤷‍♂️ It doesn't even need to match the srcset we use in the editor, this should just work well on the frontend.

Reply:
Valid points!